makefile.shade

```diff
@@ -929,13 +929,27 @@ functions
             {
                 get
                 {
-                    if (Dependencies.Count > 0)
-                    {
-                        return 1 + Dependencies.Max(d => d.Order);
-                    }
+                    return GetOrder(new HashSet<RepositoryInfo>(), this);
+                }
+            }
 
-                    return 1;
+            private static int GetOrder(HashSet<RepositoryInfo> visited, RepositoryInfo info)
+            {
+                if (!visited.Add(info))
```

Comment:
Nah, it would do reference equality. Should be fine here since we have a fixed set of instances.